src/main/java/org/elasticsearch/index/query/BoolQueryBuilder.java

```diff
@@ -200,7 +202,7 @@ private void doXArrayContent(String field, List<QueryBuilder> clauses, XContentB
     }
 
     @Override
-    protected String parserName() {
-        return BoolQueryParser.NAME;
+    public String queryId() {
+        return BoolQueryBuilder.NAME;
```

Comment:
in all the builders, you can remove the class name, just `return NAME` is enough.